### ISSUE TITLE
"timeout waiting for the application to start" -- but the app starts afterwards

### ISSUE THREAD

**Issue by filiph** [flutter/flutter#23861]
## Steps to Reproduce

The following may be dependent on the dev machine, but I have no way to test. The following happens on my `MacBook Pro (Retina, 15-inch, Mid 2014)`.

1. Open Android Studio (v3.2.1, with Flutter plugin 29.1.1 and Dart plugin 181.5656).
2. Create new Flutter app
3. Open iOS Simulator.
4. Debug/Run the app in the Simulator

The debug log shows the following:

<img width="689" alt="screen shot 2018-11-02 at 10 07 01" src="https://user-images.githubusercontent.com/919717/47905470-8bab2300-de87-11e8-8ec3-385a598b4d66.png">

Note that the "timeout waiting for the application to start" is red. It stays there for at least a few seconds (probably tens of seconds) before the build finishes, and the next line is printed out. 

After that, everything works just fine.

I'm filing this mostly because I've seen this happen to newcomers during codelabs, and they (understandably) assume that the build failed and they should just stop it.

## Logs

```
Launching lib/main.dart on iPhone X in debug mode...
Running pod install...
Starting Xcode build...
timeout waiting for the application to start
Xcode build done.                                           127.6s
Syncing files to device iPhone X...
	[C1.1 E9C3A58A-EFF3-459E-A2C8-C1D461C3C624 192.168.0.14:54760<->172.217.23.202:443]
	Connected Path: satisfied (Path is satisfied), interface: en0
	Duration: 120.981s, DNS @0.003s took 0.018s, TCP @0.024s took 0.012s, TLS took 0.170s
	bytes in/out: 3306/1028, packets in/out: 9/9, rtt: 0.025s, retransmitted packets: 0, out-of-order packets: 0
	[C2.1 E0EF4920-9679-45B3-8EDE-B64A67A08769 192.168.0.14:54785<->216.58.201.110:443]
	Connected Path: satisfied (Path is satisfied), interface: en0
	Duration: 120.488s, DNS @0.000s took 0.001s, TCP @0.003s took 0.010s, TLS took 0.108s
	bytes in/out: 3876/1023, packets in/out: 10/9, rtt: 0.022s, retransmitted packets: 1, out-of-order packets: 0
	[C3.1 9B325582-F53A-415B-8FAF-BBAFC43598B7 192.168.0.14:54786<->216.58.201.110:443]
	Connected Path: satisfied (Path is satisfied), interface: en0
	Duration: 121.090s, DNS @0.000s took 0.002s, TCP @0.004s took 0.015s, TLS took 0.121s
	bytes in/out: 3588/1165, packets in/out: 9/9, rtt: 0.026s, retransmitted packets: 0, out-of-order packets: 0
[...]
```

```
$ flutter analyze
Analyzing baby_names...
No issues found! (ran in 2.7s)
```

```
$ flutter doctor -v
[✓] Flutter (Channel beta, v0.9.4, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.9.4 at /Users/filiph/dev/flutter
    • Framework revision f37c235c32 (5 weeks ago), 2018-09-25 17:45:40 -0400
    • Engine revision 74625aed32
    • Dart version 2.1.0-dev.5.0.flutter-a2eb050044

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/filiph/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 29.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[!] IntelliJ IDEA Community Edition (version 2018.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.28.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.19.0

[✓] Connected devices (1 available)
    • iPhone X • A4DBE660-4C39-4AEA-B258-7755F584BA6E • ios • iOS 12.0 (simulator)

! Doctor found issues in 1 category.
```

**Comment by jayantakumar** [comment]
i too got the same please help

**Comment by jayantakumar** [comment]
try flutter clean ,restart your device i made a new project flutter clean ran it and it magically worked

**Comment by DanTup** [comment]
I think this may be the same as #22338. There's a timeout in the startup code which behaves badly (when the timeout is hit, it carries on as if the thing that timed out completed, but since it didn't, strange things can happen).

(@Hixie is working on removing timeouts on a branch, but I think it may be blocked - I think it may be worth removing this timeout on its own to resolve this? It was relatively recently in #21696)

**Comment by RonenR** [comment]
Same here. Found a workaround:
`flutter build ios --debug -v > build.log`
And then run the app itself using Xcode.
It seems to me it's similar to issues with release as well.

**Comment by zoechi** [comment]
I think this was fixed by #24336
Please reopen if you are still seeing this.
